server/broadcast_test.go

```diff
@@ -1033,3 +983,118 @@ func TestVerifier_HLSInsertion(t *testing.T) {
+	externalOS := &stubOSSession{}
+	bsm := bsmWithSessList([]*BroadcastSession{})
+	cxn := &rtmpConnection{
+		mid:         mid,
+		pl:          &stubPlaylistManager{manifestID: mid},
+		profile:     &ffmpeg.P240p30fps16x9,
+		sessManager: bsm,
+	}
+	seg := &stream.HLSSegment{}
+
+	oldDownloadSeg := downloadSeg
+	defer func() { downloadSeg = oldDownloadSeg }()
+
+	downloaded := make(map[string]bool)
+	downloadSeg = func(url string) ([]byte, error) {
+		downloaded[url] = true
+
+		return []byte("foo"), nil
+	}
+
+	//
+	// Tests when there is no verification policy
+	//
+
+	// When there is no broadcaster OS, segments should not be downloaded
+	url := "somewhere1"
+	cxn.sessManager = bsmWithSessList([]*BroadcastSession{genBcastSess(t, url, nil, mid)})
+	_, err := transcodeSegment(cxn, seg, "dummy", nil)
+	assert.Nil(err)
+	assert.False(downloaded[url])
+
+	// When segments are in the broadcaster's external OS, segments should not be downloaded
+	url = "https://livepeer.s3.amazonaws.com/resp1"
+	cxn.sessManager = bsmWithSessList([]*BroadcastSession{genBcastSess(t, url, externalOS, mid)})
+	_, err = transcodeSegment(cxn, seg, "dummy", nil)
+	assert.Nil(err)
+	assert.False(downloaded[url])
+
+	// When segments are not in the broadcaster's external OS, segments should be downloaded
+	url = "somewhere2"
+	cxn.sessManager = bsmWithSessList([]*BroadcastSession{genBcastSess(t, url, externalOS, mid)})
+	_, err = transcodeSegment(cxn, seg, "dummy", nil)
+	assert.Nil(err)
+	assert.True(downloaded[url])
+
+	//
+	// Tests when there is a verification policy
+	//
+
+	verifier := newStubSegmentVerifier(&stubVerifier{retries: 100})
+
+	// When there is no broadcaster OS, segments should be downloaded
+	url = "somewhere3"
+	cxn.sessManager = bsmWithSessList([]*BroadcastSession{genBcastSess(t, url, nil, mid)})
+	_, err = transcodeSegment(cxn, seg, "dummy", verifier)
+	assert.Nil(err)
+	assert.True(downloaded[url])
+
+	// When segments are in the broadcaster's external OS, segments should be downloaded
+	url = "https://livepeer.s3.amazonaws.com/resp2"
+	cxn.sessManager = bsmWithSessList([]*BroadcastSession{genBcastSess(t, url, externalOS, mid)})
+	_, err = transcodeSegment(cxn, seg, "dummy", verifier)
+	assert.Nil(err)
+	assert.True(downloaded[url])
+
+	// When segments are not in the broadcaster's exernal OS, segments should be downloaded
+	url = "somewhere4"
+	cxn.sessManager = bsmWithSessList([]*BroadcastSession{genBcastSess(t, url, externalOS, mid)})
+	_, err = transcodeSegment(cxn, seg, "dummy", verifier)
+	assert.Nil(err)
+	assert.True(downloaded[url])
+}
+
+func genBcastSess(t *testing.T, url string, os drivers.OSSession, mid core.ManifestID) *BroadcastSession {
+	segData := []*net.TranscodedSegmentData{
+		{Url: url, Pixels: 100},
+	}
+
+	buf, err := proto.Marshal(&net.TranscodeResult{
+		Result: &net.TranscodeResult_Data{
+			Data: &net.TranscodeData{Segments: segData},
+		},
+	})
+	assert.Nil(t, err, fmt.Sprintf("Could not marshal results for %s", url))
```

Comment:
nit: shouldn't this be `require.Nil` ?

Reply:
Yep - the assert statement was copy and pasted. Fixed in 75f9d11